utils/CMakeLists.txt

```diff
@@ -0,0 +1,41 @@
+# enable utils build a common library for all utils
+set(QUDA_UTILS_COMMON misc.cpp face_gauge.cpp host_utils.cpp host_blas.cpp command_line_params.cpp llfat_utils.cpp staggered_gauge_utils.cpp set_params.cpp)
+cuda_add_library(quda_utils STATIC ${QUDA_UTILS_COMMON})
+if(QUDA_QMP AND QUDA_DOWNLOAD_USQCD AND NOT QUDA_QMPHOME)
+  add_dependencies(quda_utils QMP)
+endif()
+
+if(QUDA_QDPJIT)
+  target_link_libraries(quda
+    INTERFACE ${QDP_LDFLAGS} ${QDP_LIB} ${QDP_LIBS} ${QIO_LIB} ${LIME_LIB} ${QUDA_QMP_LDFLAGS}
+    ${QMP_LIB} ${MPI_CXX_LIBRARIES})
+endif()
+
+if(QUDA_QIO AND QUDA_DOWNLOAD_USQCD AND NOT QIO_FOUND)
```

Comment:
As this shows up in many places, can we make sure to only add it where it is actually used. So only if it actually is a dependency. (Must admit I have not verified this yet)